scripts/quality/check_codacy_zero.py

```
@@ -0,0 +1,163 @@
+    parser = argparse.ArgumentParser(description="Assert Codacy has zero total open issues.")
+    parser.add_argument("--provider", default="gh")
+    parser.add_argument("--owner", required=True)
+    parser.add_argument("--repo", required=True)
+    parser.add_argument("--token", default="")
+    parser.add_argument("--out-json", default="codacy-zero/codacy.json")
+    parser.add_argument("--out-md", default="codacy-zero/codacy.md")
+    return parser.parse_args()
+
+
+def _request_json(url: str, token: str, *, method: str = "GET", data: dict[str, Any] | None = None) -> dict[str, Any]:
+    body = json.dumps(data).encode("utf-8") if data is not None else None
```

Ruff (0.15.5)

[error] Undefined name `json`

(F821)